tests/tracer/test_span.py

```diff
@@ -138,17 +137,11 @@ def __repr__(self):
 
     def test_finish(self):
         # ensure finish will record a span
-        ctx = Context()
-        s = Span(self.tracer, "test.span", context=ctx)
-        ctx.add_span(s)
-        assert s.duration is None
-
+        s = Span(self.tracer, "test.span")
         sleep = 0.05
-        with s as s1:
-            assert s is s1
-            time.sleep(sleep)
+        time.sleep(sleep)
+        s.finish()
         assert s.duration >= sleep, "%s < %s" % (s.duration, sleep)
-        self.assert_span_count(1)
```

Comment:
Nitpicking, but can we also check that the span has been recorded? Is there a reason the `self.assert_span_count(1)` was removed?

Reply:
yes, good catch! the span should still be generated in this case so we should be able to keep the assertion :+1:

Reply:
oh, actually on second thought the `Context` is required here 😞 so the span won't be recorded

Reply:
Oh okay, makes sense... Can we then update the test's docstring to be a bit more specific?
I.e. `# ensure span.finish() marks the end time of the span and can be recorded`

Reply:
sure, we can also remove the `tracer` usage since it's not necessary for this test

Reply:
Yeah, I'm good with that as well :)